FOUNDATIONS.md

```diff
@@ -13,15 +13,12 @@ ensure the success of future improvements (especially type systems).
 
 ## Linked
 
-IPLD must support linking to any IPLD node (even if the node is in the middle of
-a block). That is, IPLD must support arbitrary IPLD paths in links.
+The IPLD Data Model includes Links. A Link can be resolved to reach another IPLD Node.
 
-**Motivation:** Considering this in the context of programming languages, not
-being able to *store* a pointer to a struct *inside* of another struct would be
-severely limiting.
-
-NOTE: We don't currently support arbitrary paths but, in the context of
```

Comment:
I don't understand this, "arbitrary paths". By removing it are we missing something important?

Reply:
Assuming that it's correct to parse that sentence as a continuation of the thought above about links which aim directly to mid-block nodes, my refutation is is here: https://github.com/ipld/specs/pull/146#discussion_r308827938 -- in short, that we can't actually act as if this is fully and clearly defined; it isn't.

Reply:
`/ipld/QmFoobar/path/to/thing` is an arbitrary IPLD path. The requirement is to be able to use any arbitrary path as a link in an IPLD object.

Reply:
So this is the broader topic of is a Link as CID or CID + Path. I think we agree that this needs further discussion. Could we perhaps re-add the TODO section of this document and put in that issue? I really don't want to have the overall document blocked on that discussion, which I think is a pretty big one.

Reply:
I'm slowly becoming more convinced this is at least viable by further discussion over in https://github.com/ipld/specs/pull/146#discussion_r310810675 -- specifically, someone suggesting that implementing updates to structures by copying data from a CID+Path link to a new block would be considered acceptable -- but agree that this still needs a lot more discussion -- for example, how many people would consider that copy-to-a-new-block behavior wise or unsurprising?  If that implication, and any further transitive implications it has, are surprising enough to enough people, is there a point where that means it's a feature we actually don't want to pursue after all?

This is also discussed in #145, and [I'm singing the same tune there](https://github.com/ipld/specs/issues/145#issuecomment-513585077); it's really problematic to say "we're definitely doing X" for any value of "X" that isn't fully explored nor even partially prototyped.  A commitment to "linking" in general is one thing and we all agree on it; a commitment to these particular features of linking seems... well, maybe.

I'd be supportive of adding back in a TODO so it's clear from within this document that we have work to do to resolve this.  I'd also be receptive to putting text about mid-block linking back here in the top if we had a really good set of texts to describe what that implies, but I don't know if that's something that's easy to do in this PR; I'd rather do this narrowing first, and then have an expansion back to CID+Path/midblock-targets be proposed afterward.